pkg/controller/dynamic/dynamic_controller_integration_test.go

```diff
@@ -367,9 +371,6 @@ func testUpdate(ctx context.Context, t *testing.T, testContext testrunner.TestCo
 
 	// Update resource from test data
 	updateUnstruct := testContext.UpdateUnstruct.DeepCopy()
-	if updateUnstruct == nil {
-		t.Fatalf("updateUnstruct is nil for '%v'. should SkipUpdate be set to true in resourcefixture/contexts?", testContext.ResourceFixture.Name)
-	}
```

Comment:
It seems like there's a plan to move away from `SkipUpdate` so disregard if this point will become moot in the future but as I onboarded on the code base, what I liked about this failure is that it gave me a strong signal for how to set up the fixtures tests.

Reply:
We probably need a better way to ensure coverage. This change isn't really part of it.

Regarding testdata, if we do want to enforce the existence of update.yaml, we probably should do it via a unit test but not the integration test.